scheduler/reconcile.go

```diff
@@ -1008,11 +1030,13 @@ func (a *allocReconciler) computeStopByReconnecting(untainted, reconnecting, sto
 
 	for _, reconnectingAlloc := range reconnecting {
 		// if the desired status is not run, or if the user-specified desired
-		// transition is not run, stop the allocation.
+		// transition is not run, stop the reconnecting allocation.
 		if reconnectingAlloc.DesiredStatus != structs.AllocDesiredStatusRun ||
 			reconnectingAlloc.DesiredTransition.ShouldMigrate() ||
 			reconnectingAlloc.DesiredTransition.ShouldReschedule() ||
-			reconnectingAlloc.DesiredTransition.ShouldForceReschedule() {
+			reconnectingAlloc.DesiredTransition.ShouldForceReschedule() ||
+			reconnectingAlloc.Job.Version < a.job.Version ||
```

Comment:
Should these just be != instead of < ?  Is a rollback possible?  Same question for all these checks.

Reply:
I definitely _shouldn't_ be possible to roll back a version (a rolled-back deployment always creates a new version), but having the `<` assertion seems like it gives us some guard rails for sure. I'd keep it like this.